src/components/TodoTask.jsx

```diff
@@ -0,0 +1,38 @@
+import React, {useState} from 'react';
+import EditingTemplate from './EditingTemplate';
+import ViewTemplate from './ViewTemplate';
+
+export default function TodoTask({id, text, onDeleteTask, onEditTask}) {
+    
+    const [isEditing, setEditing] = useState(false);
+
+    function hideEditingTemplate () {
+        setEditing(false);
+    }
+
+    function showEditingTemplate () {
+        setEditing(true);
+    }
+
+    return (
+        <>
+            <li>
+                {isEditing ? (
+                    <EditingTemplate
+                        id={id}
+                        previousTask={text}
+                        onEditTask={onEditTask}
+                        onHideEditingTemplate={hideEditingTemplate}
+                    />
```

Comment:
Având în vedere că dai toate props-urile mai departe, ai fi putut să faci:

```js
export default function TodoTask(props) {
```
```suggestion
                    <EditingTemplate {...props} onHideEditingTemplate={hideEditingTemplate} />
```